src/SFA.DAS.ApprenticeCommitments/DTOs/ApprenticeshipDtoMapping.cs

```diff
@@ -57,8 +60,45 @@ private static ApprenticeshipDto MapApprenticeshipAndRevisionToApprenticeshipDto
                 ApprovedOn = latest.CommitmentsApprovedOn,
                 ChangeOfCircumstanceNotifications = apprenticeship.ChangeOfCircumstanceNotifications,
                 StoppedReceivedOn = latest.StoppedReceivedOn,
-                HasBeenConfirmedAtLeastOnce = apprenticeship.ApprenticeshipHasPreviouslyBeenConfirmed
+                HasBeenConfirmedAtLeastOnce = apprenticeship.ApprenticeshipHasPreviouslyBeenConfirmed,
+                Revisions = GetRevisions(apprenticeship.Revisions),
             };
         }
+
+        private static List<RevisionDto> GetRevisions(IReadOnlyCollection<Revision> revisions)
+        { 
+            var ret = new List<RevisionDto>();
+            if (revisions.Count == 1)
+                return ret;
+
+            var rev = revisions.OrderByDescending(x => x.Id).Pairwise((x, y) =>
+            {
+                if (x.Details.DeliveryModel != y.Details.DeliveryModel)
+                    ret.Add(new RevisionDto("Delivery model changed",                                                
+                         $"Delivery model changed from {y.Details.DeliveryModel} to {x.Details.DeliveryModel}",
+                         x.CreatedOn));
+
+                if (x.Details.EmployerName != y.Details.EmployerName)
+                    ret.Add(new RevisionDto("You started with a new employer",
+                        $"Employer changed from {y.Details.EmployerName} to {x.Details.EmployerName}",
+                        x.CreatedOn));
+
+                if (x.Details.TrainingProviderName != y.Details.TrainingProviderName)
+                    ret.Add(new RevisionDto("You started with a new training provider",
+                        $"Provider changed from {y.Details.TrainingProviderName} to {x.Details.TrainingProviderName}",
+                        x.CreatedOn));
+
+                return x;
+            }).ToList();
+
+            var confirmedOn = revisions.FirstOrDefault(x => x.ConfirmedOn.HasValue)?.ConfirmedOn;
+
+            if (confirmedOn.HasValue && ret.Count > 0)
+                ret.Add(new RevisionDto("You started your apprenticeship",
+                    "You confirmed your apprenticeship details",
+                    confirmedOn));
+
+            return ret;
+        }
```

Comment:
The revisions here should only be the revisions prior to the last confirmed apprenticeship. As anything after not will still need to be confirmed.